backend/src/main/java/com/empire/World.java

```diff
@@ -761,33 +761,50 @@ void setGothiSpellStatus() {
 				}
 				if (voting) getNation(k).goodwill -= 30;
 			}
+			// Spell tickers.
 			if (passedSpells.contains("Rjinku")) {
-				if (!tivar.quake)	notifyAllPlayers("The Quake Begins", "The necessary number of gothi of Rjinku have agreed to call forth the Quake. Buildings and crops will be destroyed every week until the earthquakes end.");
-				tivar.quake = true;
-			} else if (tivar.quake) {
-				tivar.quake = Math.random() < 0.5;
-				if (!tivar.quake) notifyAllPlayers("Earthquakes End", "The magical earthquakes wracking the land have finally been ended.");
+				if (tivar.quake == 0)	notifyAllPlayers("The Quake Begins", "The necessary number of gothi of Rjinku have agreed to call forth the Quake. Buildings and crops will be destroyed every week until the earthquakes end.");
+				tivar.quake++;
+			} else if (tivar.quake > 0) {
+				if (Math.random() > Constants.tivarSpellContinueChance) {
+					tivar.quake = 0;
+					notifyAllPlayers("Earthquakes End", "The magical earthquakes wracking the land have finally been ended.");
+				} else {
+					tivar.quake++;
+				}
 			}
 			if (passedSpells.contains("Syrjen")) {
-				if (!tivar.deluge) notifyAllPlayers("The Deluge Begins", "The necessary number of gothi of Syrjen have agreed to call forth the Deluge. Terrible floods make all land regions treacherous (if navigable by fleets), and more crops are drowned in the muddy waters every week.");
-				tivar.deluge = true;
-			} else if (tivar.deluge) {
-				tivar.deluge = Math.random() < 0.5;
-				if (!tivar.deluge) notifyAllPlayers("Deluge Ends", "The magical deluge drowning the land has finally ceased.");
+				if (tivar.deluge == 0) notifyAllPlayers("The Deluge Begins", "The necessary number of gothi of Syrjen have agreed to call forth the Deluge. Terrible floods make all land regions treacherous (if navigable by fleets), and more crops are drowned in the muddy waters every week.");
+				tivar.deluge++;
+			} else if (tivar.deluge > 0) {
+				if (Math.random() > Constants.tivarSpellContinueChance) {
+					tivar.deluge = 0;
+					notifyAllPlayers("Deluge Ends", "The magical deluge drowning the land has finally ceased.");
+				} else {
+					tivar.deluge++;
+				}
 			}
 			if (passedSpells.contains("Lyskr")) {
-				if (!tivar.veil) notifyAllPlayers("The Veil Falls", "The necessary number of gothi of Lyskr have agreed to call forth the Veil. Heavy fog chokes out the sun and reduces visibility severely. Barred from the sunlight, crops will be lost every week until the fog lifts.");
-				tivar.veil = true;
-			} else if (tivar.veil) {
-				tivar.veil = Math.random() < 0.5;
-				if (!tivar.veil) notifyAllPlayers("Veil Ends", "The magical fog blanketing the land has finally lifted.");
+				if (tivar.veil == 0) notifyAllPlayers("The Veil Falls", "The necessary number of gothi of Lyskr have agreed to call forth the Veil. Heavy fog chokes out the sun and reduces visibility severely. Barred from the sunlight, crops will be lost every week until the fog lifts.");
```

Comment:
This description seems to contradict the idea of the crop-destroying grace period.

Reply:
Done.